libraries/classes/UrlRedirector.php

```diff
@@ -17,7 +17,7 @@ final class UrlRedirector
     /**
      * @psalm-return never
      */
-    public static function redirect(): void
+    public static function redirect(?string $url): void
```

Comment:
```suggestion
    public static function redirect(string $url): void
```
It doesn't need to be nullable. Why not just provide an empty string instead? Then all the `(string)` can be removed and the check for `is_scalar` or `=== null` be removed too.